Refrigerator/Refrigerator/Refrigerator.cs

```diff
@@ -0,0 +1,184 @@
+            }
+            else
+                throw new Exception("The item is already in a refrigerator.");
+        }
+
+        public void RemoveItem(int itemID)
+        {
+            var flag = false;
+            var shelfNumber = 0;
+            var item = Item.GetItemByID(itemID);
+            for (int i = 0; i < NumberOfShelves; i++)
+            {
+                if (Shelves[i].IsItemInShelf(item))
+                {
+                    flag = true;
+                    shelfNumber = i;
+                }
+            }
+            if (flag)
+                Shelves[shelfNumber].RemoveItem(item);
+            else
+                throw new Exception("The refrigerator does not contain such an item.");
+        }
+
+        public List<Item> CleanRefrigerator()
+        {
+            var itemsToTrowAway = new List<Item>();
+            foreach (var shelf in Shelves)
+                itemsToTrowAway.AddRange(shelf.CleanShelf());
+            return itemsToTrowAway;
+        }
+
+        public List<Item> FeelLikeEating(KosherType kosher, ItemType type)
+        {
+            var items = new List<Item>();
+            foreach (var shelf in Shelves)
+            {
+                var itemsToAdd = shelf.GetItemByKosherAndType(kosher, type);
+                if (itemsToAdd != null)
+                    items.AddRange(itemsToAdd);
+            }
+            return items;
+        }
+
+        public void GettingReadyForShopping()
+        {
+            if (HowMuchSpaceIsLeft() < 20)
+            {
+                CleanRefrigerator();
+                if (HowMuchSpaceIsLeft() < 20)
+                {
+                    double takeSpace = 0;
+                    double spaceByKosherAndExpiryDate;
+                    List<Item> itemsToThrowAway = new List<Item>();
+                    List<Item> itemsByKosherAndExpiryDate = new List<Item>();
+                    foreach (var shelf in Shelves)
+                    {
+                        (itemsByKosherAndExpiryDate, spaceByKosherAndExpiryDate) = shelf.CleanByKosherAndExpiryDate();
+                        takeSpace += spaceByKosherAndExpiryDate;
+                        itemsToThrowAway.AddRange(itemsByKosherAndExpiryDate);
+                    }
+                    if (HowMuchSpaceIsLeft() - takeSpace < 20)
+                    {
+                        foreach (var shelf in Shelves)
+                        {
+                            (itemsByKosherAndExpiryDate, spaceByKosherAndExpiryDate) = shelf.CleanByKosherAndExpiryDate();
+                            shelf.RemoveItems(itemsByKosherAndExpiryDate);
+                        }
+                        Console.WriteLine("Cleaning has been done for non-expired products, shopping can be done");
+                    }
+                    else
+                        Console.WriteLine("There is not enough space in the refrigerator, this is not the time to go shopping");
```

Comment:
Bug: you don't take into consideration the preference of items to throw (first milk, then meat then parve)